client/functions.lua

```diff
@@ -62,34 +71,54 @@ end
 
 function public.attemptPoliceAlert(type)
     if not alertSend then
+        alertSend = true
         local chance = config.policeAlertChance
+
         if GetClockHours() >= 1 and GetClockHours() <= 6 then
             chance = config.policeNightAlertChance
         end
+
         if math.random() <= chance then
             TriggerServerEvent('police:server:policeAlert', locale("info.vehicle_theft") .. type)
         end
-        alertSend = true
+
         SetTimeout(config.alertCooldown, function()
             alertSend = false
         end)
     end
 end
 
 --- Gets bone coords
---- @param vehicle number The entity number of the vehicle.
+--- @param entity number The entity index.
 --- @param boneName string The entity bone name.
 --- @return vector3 Bone coords if exists, entity coords otherwise.
-local function getBoneCoords(vehicle, boneName)
-    local boneIndex = GetEntityBoneIndexByName(vehicle, boneName)
+local function getBoneCoords(entity, boneName)
+    local boneIndex = GetEntityBoneIndexByName(entity, boneName)
 
     if boneIndex ~= -1 then
-        return GetWorldPositionOfEntityBone(vehicle, boneIndex)
+        return GetWorldPositionOfEntityBone(entity, boneIndex)
     else
-        return GetEntityCoords(vehicle)
+        return GetEntityCoords(entity)
     end
 end
 
+--- comment
+--- @param coords vector3
+--- @param entity number
+--- @param bones table
+--- @param maxDistance number
+--- @return boolean? Bone coords if exists, entity coords otherwise.
+local function isCloseToAnyBone(coords, entity, bones, maxDistance)
+    for _, boneName in ipairs(bones) do
```

Comment:
prefer numeric for loops to iterate over arrays https://manason.github.io/effective-fivem-lua/tables/#use-numeric-for-loops-when-iterating-over-an-array

Here is the benchmarking around different for loop types: https://gist.github.com/thelindat/939fb0aef8b80a077f76f1a850b2a53d?permalink_comment_id=4257719#gistcomment-4257719

Reply:
a lot of useful information, thanks!

Reply:
I'm still not sure it's worth putting performance over readability here

```lua
for _, v in ipairs(config.noCarjackWeapons) do
   if weapon == joaat(v) then return true end
end
```

```lua
for i = 1, #config.noCarjackWeapons do
   if weapon == joaat(config.noCarjackWeapons[i]) then return true end
end
```